bot/mc.go

```diff
@@ -4,43 +4,137 @@ import (
+			if err != nil {
+				log.Printf("error editting message inside of manageMiencraftCmd %s", err)
+			}
+		}()
+	}
+	return nil
+}
+
+func mcCmd(ctx CommandCtx) error {
+	m := ctx.Message
+	if m.Author.ID == blueId || m.Author.ID == morthisId {
+		res, err := minecraftCommand(strings.Join(ctx.Args, " "))
+		if err != nil {
+			log.Printf("Err: %s", err)
+			return ctx.Reply("Could not send command, Minecraft might be offline.")
+		}
+		if res == "" {
+			return nil
+		}
+		return ctx.Reply(res)
+	}
+	return ctx.Reply("Sorry you're not a Minecraft server admin.")
+
+}
+
+func execDockerCompose(action string) error {
 	client, err := sshcmd.ConnectToDev()
 	if err != nil {
-		log.Printf("SSH error: %v", err)
-		return
+		return fmt.Errorf("ssh error: %w", err)
 	}
 	defer client.Close()
 
-	com, err := sshcmd.RunCommand(client, "docker-compose -f /app/docker-compose.yml restart")
+	com, err := sshcmd.RunCommand(client, fmt.Sprintf("docker-compose -f /app/docker-compose.yml %s", action))
 	if err != nil {
-		log.Println("Error:", err.Error())
+		return fmt.Errorf("error running ssh command: %w", err)
 	}
 	log.Printf("Compose command response: %s", com)
+	return nil
 }
-
-func minecraftCommand(command string) (string, error) {
-
+func initMC() (string, string, string, error) {
 	rconHost := os.Getenv("MC_HOST")
 	rconPort := os.Getenv("RCON_PORT")
 	rconPass := os.Getenv("RCON_PASS")
 
-	con, err := rcon.Dial(fmt.Sprintf("%s:%s", rconHost, rconPort), rconPass)
+	if rconHost == "" || rconPort == "" || rconPass == "" {
+		err := fmt.Errorf("missing rcon env vars")
+		log.Printf("Error: %v", err)
+		return "", "", "", err
+	}
+	return rconHost, rconPort, rconPass, nil
+}
+
+func minecraftCommand(command string) (string, error) {
+	host, port, pass, err := initMC()
 	if err != nil {
-		log.Printf("Error, %s:", err)
 		return "", err
 	}
+	log.Printf("Cmd here %s", command)
+	con, err := rcon.Dial(fmt.Sprintf("%s:%s", host, port), pass)
+	if err != nil {
+		log.Printf("Error connecting to RCON: %s", err)
+		return "", fmt.Errorf("unable to connect to rcon server: %s", err)
+	}
 	defer con.Close()
-	response, err := con.Execute(command)
+	resp, err := con.Execute(command)
 	if err != nil {
 		log.Printf("Error, %s", err)
 		return "", err
 	}
-	log.Printf("Response %s", response)
-	return response, nil
+	return resp, nil
+}
+
+func whitelistCmd(ctx CommandCtx) error {
+	msg := ctx.Message.Message
+	args := ctx.Args
+	if memberHasRole(msg.Member, mcRoleId) || msg.Author.ID == morthisId {
+		log.Printf("Whitelisting, %s ", args[0])
+		res, err := minecraftCommand(fmt.Sprintf("whitelist add %s", args[0]))
+		if err != nil {
+			if err := ctx.Reply("Could not send command, Minecraft might be offline."); err != nil {
+				return err
+			}
+		}
+		if err = ctx.Reply(res); err != nil {
+			return err
+		}
+		if err := ctx.Reply("You're not a minecraft admin."); err != nil {
+			return err
+		}
```

Comment:
What in gods name is going on here?

Reply:
The linter was complaining about not handling errors